patch/go.proto

```diff
@@ -30,22 +30,41 @@ message Options {
 	optional string stringer_name = 30; // TODO: implement for messages
 }
 
-extend google.protobuf.MessageOptions {
-	optional Options message = 7001;
+// File-level options
+message FileOptions {
+	// The disable_enum_prefixes option disables adding a prefix with a type name for an enum value.
+	optional bool disable_enum_prefixes = 1;
+
+	// The fix_initialisms option changes message field names to the golang naming convention.
+	// So for example the field named in proto file "id" would be named "ID" in go.
+	optional bool fix_initialisms = 2;
+
+	// The initialisms option contains the list of initialisms used if the fix_initialisms option is enabled.
+	// Initialisms should be in uppercase. This list may be empty, in this case the default golint initialisms
+	// list will be used.
+	repeated string initialisms = 3;
+}
```

Comment:
I’d imagined this as a nested set of options, so one could do something like this:

```proto
option (go.initialisms) = 'ID';
option (go.initialisms) = 'URL';
option (go.values) = {
    prefix: ''
};
```

I pushed a spike design here: https://github.com/alta/protopatch/blob/ydnar/file-options-design/patch/go.proto

Reply:
Additionally, it would not have a default list of initialisms from `golint`, but explicitly work off the initialisms you specify in the file.

Reply:
> I’d imagined this as a nested set of options, so one could do something like this:

It is possible now. You can write:
```
option (go.file) = {
    fix_initialisms: true,
    initialisms: ["ID", "URL"]    
};
```
or:
```
option (go.file).fix_initialisms = true;
option (go.file).initialisms = "ID";
option (go.file).initialisms = "URL";

```

> Additionally, it would not have a default list of initialisms from golint, but explicitly work off the initialisms you specify in the file.

Do you want to drop the default initialism list entirely? 
I think the default initialisms list is a good idea. Why? Because by enabling one option, we get a file compatible with go lint. This is hard to do by adding initialisms one by one: go lint has about 40 initialisms ...

Reply:
The default initialisms list, while convenient, isn’t exhaustive. A single proto file will have a handful of these initialisms, and I think it’s better to have each file define what exactly it wants rather than depend on a predefined list of English-centric initialisms.

I think we should favor clarity over brevity.

Note in my branch I skipped `go.file` in favor of `go.messages`, `go.enums`, etc. to illustrate where each change would apply, to all messages, fields, enums, values, and oneofs.

The one issue with this is that having a top-level `FileOptions` extension `initialisms` means that this isn’t possible:

```proto
option (go.initialisms) = ["ID", "URL"];
```

Reply:
I guess we could nest our own `FileOptions` as a single extension to `FileOptions`, with `messages`, `enums`, `fields`, `values`, `oneofs`, so this would be possible:

```proto
option (go.file).values = {
    prefix: ''
}
```

Reply:
> The default initialisms list, while convenient, isn’t exhaustive. A single proto file will have a handful of these initialisms, and I think it’s better to have each file define what exactly it wants rather than depend on a predefined list of English-centric initialisms.
> 
> I think we should favor clarity over brevity.

Ok, but what if we have many proto files in the project? I have about 30 proto files in my small project. So in each file I have to define a list of the same initialisms. It is not very convenient to maintenance. How to solve it? Global option for all patched files?

Reply:
How many initialisms does each proto file use?

Reply:
In my case, not so much, probably max 3.

Reply:
@ydnar Can you look on last commit ? Is it closer to your idea?